kratos/spatial_containers/bounding_volume_tree.h

```diff
@@ -270,11 +270,11 @@ class _6DOP : public kDOP // AABB
     KRATOS_CLASS_POINTER_DEFINITION(_6DOP);
     _6DOP() : kDOP() {this->Initialize();}
     virtual ~_6DOP() {}
-    virtual std::size_t NumberOfDirections() const {return 3;}
+    virtual std::size_t NumberOfDirections() const override {return 3;}
 
 private:
     static const double msDirection[][3];
-    virtual const double (*Direction() const)[3];
+    virtual const double (*Direction() const override)[3];
```

Comment:
@vryy I see, then maybe we can just use:

```C++ 
typedef double ** const ArrayType;
```

This should be equivalent to const pointer to a array of doubles (not sure if the `const` goes there or after `double`). This way you even may use the same function in the base class without having to derive and implement a different one for every case.